diem-node/src/lib.rs

```diff
@@ -303,10 +303,18 @@ pub fn setup_environment(node_config: &NodeConfig, logger: Option<Arc<Logger>>)
         network_configs.push(network_config);
     }
 
-    let mut network_builders = Vec::new();
-
     // Instantiate every network and collect the requisite endpoints for state_sync, mempool, and consensus.
     for (idx, network_config) in network_configs.into_iter().enumerate() {
+        debug!("Creating runtime for {}", network_config.network_id);
+        let runtime = Builder::new_multi_thread()
+            .thread_name(format!("network-{}", network_config.network_id))
+            .enable_all()
+            .build()
+            .expect("Failed to start runtime. Won't be able to start networking.");
+
```

Comment:
I've long wondered if we should really just have a single runtime for an entire node... maybe this could be a follow exploration for the diem state v2 blahblah? I think the runtimes end up competing with each other.

Reply:
This is only good if we make sure we don't do `thread::sleep` anywhere

Reply:
I think that it would be a big benefit structurally if we could support having everything in a single runtime vs having tons of runtimes. It doesn't mean that we wouldn't still have the ability to have runtimes per component but it would give us a lot more flexibility wrt testing too